pylint/lint/report_functions.py

```diff
@@ -2,21 +2,24 @@
 # For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
 
 import collections
+from typing import Any, DefaultDict, Dict, Tuple
 
 from pylint import checkers, exceptions
-from pylint.reporters.ureports import nodes as report_nodes
+from pylint.reporters.ureports.nodes import Section, Table
 
 
-def report_total_messages_stats(sect, stats, previous_stats):
+def report_total_messages_stats(
```

Comment:
This is all handled by https://github.com/PyCQA/pylint/pull/4973/ and is more specific.

Reply:
Do you think we should merge 4973 first ?

Reply:
Yes! Although I'm biased because it is my own PR I think it includes some typing that is more specific than what is currently in this PR.